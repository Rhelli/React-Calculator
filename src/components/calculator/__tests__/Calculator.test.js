import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('The appearance of the calculator', () => {
  it('Matches the Calculator snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Contains the navbar as the first component on the page', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild.firstChild.classList.contains('navbarContainer')).toBe(true);
  });

  it('Contains the Calculator component as the second component on the page', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild.lastChild.classList.contains('calculatorContainer')).toBe(true);
  });

  it('Contains the calculator blurb text on the page', () => {
    const { getByText } = render(<Calculator />);
    getByText("Let's Math It Up Nerds");
  });

  it('Contains the actual calculator itself', () => {
    const { container } = render(<Calculator />);
    expect(container.firstChild.lastChild.lastChild.classList.contains('calculator')).toBe(true);
  });
});
