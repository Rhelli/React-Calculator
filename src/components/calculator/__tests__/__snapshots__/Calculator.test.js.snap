// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance of the calculator Matches the Calculator snapshot 1`] = `
<div>
  <div
    className="navbarContainer"
  >
    <div
      className="navbarTitle"
    >
      <h1>
        Math Magicians
      </h1>
    </div>
    <div
      className="navbarLinks"
    >
      <a
        href="/"
      >
        Home 
      </a>
      <span>
         | 
      </span>
      <a
        href="/Calculator"
      >
        Calculator 
      </a>
      <span>
         | 
      </span>
      <a
        href="/Quote"
      >
        Quote 
      </a>
    </div>
  </div>
  <div
    className="calculatorContainer"
  >
    <div
      className="calculatorBlurb"
    >
      <h4>
        Let's Math It Up Nerds
      </h4>
    </div>
    <div
      className="calculator"
    >
      0
      <div
        className="buttonPanel"
      >
        <div
          className="row"
          id="group1"
        >
          <div
            className="button"
          >
            <button
              className="grey"
              id="AC"
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="+/-"
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="%"
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="orange"
              id="รท"
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </div>
        </div>
        <div
          className="row"
          id="group2"
        >
          <div
            className="button"
          >
            <button
              className="grey"
              id="7"
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="8"
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="9"
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="orange"
              id="x"
              onClick={[Function]}
              type="button"
            >
              x
            </button>
          </div>
        </div>
        <div
          className="row"
          id="group3"
        >
          <div
            className="button"
          >
            <button
              className="grey"
              id="4"
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="5"
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="6"
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="orange"
              id="-"
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </div>
        </div>
        <div
          className="row"
          id="group4"
        >
          <div
            className="button"
          >
            <button
              className="grey"
              id="1"
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="2"
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="3"
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="orange"
              id="+"
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </div>
        </div>
        <div
          className="row"
          id="group5"
        >
          <div
            className="zeroButton"
          >
            <button
              className="grey"
              id="0"
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="grey"
              id="."
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </div>
          <div
            className="button"
          >
            <button
              className="orange"
              id="="
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
